tests/integration/trainingContracts.test.js

```diff
@@ -131,3 +132,72 @@ describe('COURSES ROUTES - POST /trainingcontracts', () => {
     });
   });
 });
+
+describe('TRAINING CONTRACTS ROUTES - GET /trainingcontracts', () => {
+  let authToken;
+
+  describe('TRAINING_ORGANISATION_MANAGER', () => {
+    beforeEach(populateDB);
+    beforeEach(async () => {
+      authToken = await getToken('training_organisation_manager');
+    });
+
+    it('should get course\'s training contracts', async () => {
+      const trainingContractsLength = await TrainingContract.countDocuments({ course: courseList[1]._id });
+
+      const response = await app.inject({
+        method: 'GET',
+        url: `/trainingcontracts?course=${courseList[1]._id}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(response.statusCode).toBe(200);
+      expect(response.result.data.trainingContracts.length).toEqual(trainingContractsLength);
+    });
+
+    it('should return a 404 if course doesn\'t exist', async () => {
+      const response = await app.inject({
+        method: 'GET',
+        url: `/trainingcontracts?course=${new ObjectId()}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(response.statusCode).toBe(404);
+    });
+  });
+
+  describe('Other roles', () => {
+    beforeEach(populateDB);
+
+    it('should return a 404 if user company is not attached to course and user has no vendor role', async () => {
```

Comment:
ce n'est pas plutôt l'inverse : ici on regarde le cas ou le coach est dans la structure de la formation (=> le cas 200) et plus bas on regard le cas d'un coach généralement donc une 404

Reply:
dans le other roles ont vérifie si les rôles ont accès ou pas à la route, ce n'est pas le cas ici, c'est un cas spécifique de blocage mais le coach a accès à la route, c'est donc le 200 qu'il faut mettre dans other roles

Reply:
On pourrait aussi le voir comme : le cas general c'est que le coachs n'y ont pas acces sauf si c'est bien leur formation. 
J'ai l'impression qu'on fait les deux dans le code, il faudrait statuer pour savoir lequel on veut faire generalement. Je n'ai pas trop de preference 🤔

Reply:
tu peux m'indiquer à quelle endroit on fait la deuxième méthode dans le code ça ne me dit rien du tout

Reply:
C'est un peu complique a voir donc j'ai regarde rapidement mais je crois que j'en avais vu 2 cote erp (par exemple : BILL ROUTES - GET /bills/pdfs)

Reply:
j'avoue que je préfère mon option, c'est celle qu'on utilise pour completion certificates et attendance sheets. Je trouve ça bizarre d'écrire { name: 'coach', expectedCode: 404 }, c'est toujours 200 ou 403 qu'on met ici et en se basant sur le fichier right et d'après le fichier right ici coach reçoit 200

Reply:
je pense que l'option "le cas general c'est que le coachs n'y ont pas acces sauf si c'est bien leur formation." est plus répandue dans le code, c'est comme ca qu'on faisait au début en tout cas.
Ou alors on faisait les 2 cas en dehors de la boucle for sur `roles` (dans le test sur le créneaux par exemple) ce qui est peut etre le plus claire d'ailleurs

En fait le "cas general" c'est que le coach n'a pas acces a la convention car pour ce faire il faur qu'il soit dans la meme structure ; ce qui est assez restrictif du coup. Par rapport au nombre de convention en bdd, il ne pourra acceder qu'a un tres petit nombre
C'est ce qu'on fait pour les test sur le role aidant cote ERP (quasi) tout le temps il me semble.

Dans le code il y a 3 cas : 
- on met la 403 dans la boucle for `roles.forEach((role) => { it(`should return ${role.expectedCode} as user is ${role.name}` et la 200 avant
- on met la 200 dans la boucle for `roles.forEach((role) => { it(`should return ${role.expectedCode} as user is ${role.name}` et la 403 avant
- Quand il y a 2 cas possibles, on met la 403 et la 200 en dehors de la boucle for
Les cas 1 et 3 me conviennent mais pas le 2 car pour moi il porte à confusion....

Reply:
done